notificationworkerlambda/src/main/scala/com/gu/notifications/worker/AndroidSender.scala

```diff
@@ -22,4 +31,33 @@ class AndroidSender extends SenderRequestHandler[FcmClient] {
   override val deliveryService: IO[Fcm[IO]] =
     FcmClient(config.fcmConfig).fold(e => IO.raiseError(e), c => IO.delay(new Fcm(c)))
   override val maxConcurrency = 100
+
+  //override the deliverChunkedTokens method to validate the success of sending batch notifications to the FCM client. This implementation could be refactored in the future to make it more streamlined with APNs
+  override def deliverChunkedTokens(chunkedTokenStream: Stream[IO, (ChunkedTokens, Long, Instant)]): Stream[IO, Unit] = {
+    for {
+      (chunkedTokens, sentTime, functionStartTime) <- chunkedTokenStream
+      isAllowedToSendBatch = chunkedTokens.notification.topic.forall(topic => config.allowedTopicsForBatchSend.contains(topic.toString))
```

Comment:
I initially wondered if I should create unit tests for this but the initial configuration/mocking wasn't simple. This logic would never exist in the long term, so instead I'll validate in the CODE environment (have already validated locally too).

Reply:
Sounds good to me